x-pack/plugins/apm/public/context/url_params_context/helpers.ts

```diff
@@ -19,6 +19,27 @@ function getParsedDate(rawDate?: string, options = {}) {
   }
 }
 
+export function getExactDate(rawDate?: string, options = {}) {
+  if (rawDate) {
+    const isRelativeDate = rawDate.substring(0, 3) === 'now';
+    if (isRelativeDate) {
+      const isSubtractingDate = rawDate.indexOf('-') > 0;
+      const isRoundingDate = rawDate.indexOf('/') > 0;
+
+      const rawDateWithouRounding =
+        // When relative time is subtracting a period and rounding the result (e.g. now-24h/h)
+        // removed the rounding part in order to get the exact time.
+        // This is needed because of of "Today"(now/d) and "This week"(now/w) options, it rounds the values up and down
+        // so the exact time is the rounded value.
+        isSubtractingDate && isRoundingDate
+          ? rawDate.substring(0, rawDate.indexOf('/'))
+          : rawDate;
```

Comment:
What about something like this?

```suggestion
      const rawDateWithouRounding = rawDate.replace(/\/(\w)$/, "")
```